files/image_config/kubernetes/scripts/kube_join.sh

```diff
@@ -0,0 +1,43 @@
+if [ ! -f /etc/sonic/kubeadm_join.sh ]
```

Comment:
>/etc/sonic/kubeadm_join.sh [](start = 10, length = 26)

Where can I find the code?

Reply:
This is the command generated at master node, with a token that has a lifetime. So when you are joining, you get this command from  the master and add it to switch and then run this  kube_join.sh.
This would be the action to take to join a master.

BTW, join is required just once, during which it saves required context that it would not need this command to join on subsequent reboots, until you call 'kuebadm reset'.

Reply:
I am confused with:
1. kubeadm join
2. /etc/sonic/kubeadm_join.sh
3. kube_join.sh

Could you draft a design doc with the flow? and why each is required?

---
In reply to: [404436415](https://github.com/Azure/sonic-buildimage/pull/4374#discussion_r404436415) [](ancestors = 404436415)

Reply:
The info is embedded in /usr/bin/kube_join.sh
When user run it, it gives its pre-requisites as  below.
So all user needs to know is, run kube_join.sh and he auto-learn the requirements.
Still I like the idea of design doc. BTW, where does such design doc resides ?

```
if [ ! -f /etc/sonic/kubeadm_join.sh ]
then
    echo "Run 'kubeadm token create --print-join-command' in Master."
    echo "please get join command in /etc/sonic/kubeadm_join.sh"
    exit -1
fi

if [ ! -f /etc/sonic/kube_admin.conf ]
then
    echo "please get kube context in /etc/sonic/kube_admin.conf"
    echo "It can be found @ master: /etc/kubernetes/admin.conf"
    exit -1
fi

```

Reply:
File removed